domains/transfer/Transfer.API/API.IntegrationTests/Controllers/TransferAgreementsControllerTest.cs

```diff
@@ -159,6 +159,27 @@ public async Task Create_ShouldFail_WhenReceiverTinInvalid(string tin, string ex
         validationProblemContent.Should().Contain("ReceiverTin");
     }
 
+    [Fact]
```

Comment:
This should be a Theory and you should provide sets of strings and booleans. That way you can test that it is validating correctly for invalid and valid strings. You should have some invalid strings that are close to being valid and perhaps some with weird non-printable control bytes in them, if you are feeling wild :)

Reply:
I have included some additional cases, to the now theory test. Check it out: https://github.com/Energinet-DataHub/energy-origin/pull/443/commits/10eaafe8a7a658379954d3c246e045c7bb138a5e